x-pack/platform/plugins/shared/streams/server/routes/streams/crud/read_stream.ts

```diff
@@ -139,7 +145,7 @@ export async function readStream({
     effective_lifecycle: findInheritedLifecycle(streamDefinition, ancestors),
     effective_settings: getInheritedSettings([...ancestors, streamDefinition]),
     inherited_fields: inheritedFields,
+    effective_failure_store: findInheritedFailureStore(streamDefinition, ancestors),
```

Comment:
In the default retention case (lifecycle is enabled but no explicit retention was provided), this will return an object that's not really interesting:
```
"effective_failure_store": {
  "lifecycle": {
    "enabled": {}
  },
  "from": "logs.child"
}
```

The effective data_retention can be retrieved from the data stream object, like we do for classic stream. We should do the same here for wired streams that don't set an explicit retention